bin/extract-ORFs.py

```diff
@@ -64,19 +86,59 @@ def translations(read):
         def translations(read):
             return read.translations()
 
+    if kozakInfo or kozakOnly:
+        if aa:
+            # Should the code break in this case?
+            print('Did you pass NA sequences? Kozak consensuses can only'
+                  'be found in a nucleotide sequence.', file=sys.stderr)
+        else:
+            from dark.dna import findKozakConsensus
+
+        def writeToKozakOut(kozakread, kozakInfoFile):
+            """
+            Writes out information about a Kozak sequence stored in kozakread
+            to the file name given in kozakInfoFile.
+            """
+            kozakout = open(kozakInfoFile, 'a')
+            kozakout.write('Read ID: ' + kozakread.id)
```

Comment:
I would just use `print(..., file=kozakout)`.

Reply:
And use something like

```python
with open(...) as kozakfp:
    print(..., file=kozakfp)
```

Then you don't have to close the file (or worry about what happens if the intermediate code fails in some way.

There's also a way to temporarily re-assign `sys.stdout` so you can just use `print` (with no `file=` bit). But I'm not sure that is portable to Python2 without a bit of effort. E.g., see https://docs.python.org/3.4/library/contextlib.html#contextlib.redirect_stdout

Reply:
done

Reply:
(I have to do `with open (..., 'a+') as kozakfp:` to also create the file